cookbooks/swift/recipes/source.rb

```diff
@@ -117,26 +103,39 @@
   # seems to pull back pbr to 0.11 and break everything; not installing
   # swift-bench's test-requirements is probably better than that
   command "pip install -e ."
-  if not node['full_reprovision']
-    creates "/usr/local/lib/python2.7/dist-packages/swift-bench.egg-link"
-  end
   action :run
 end
 
 execute "python-swift-install" do
   cwd "#{node['source_root']}/swift"
-  command "pip install #{if not node['use_python3'] then '-c py2-constraints.txt' end} -e .[kmip_keymaster] -r test-requirements.txt"
-  if not node['full_reprovision']
-    creates "/usr/local/lib/python2.7/dist-packages/swift.egg-link"
-  end
+  command "pip install -e .[kmip_keymaster] -r test-requirements.txt"
```

Comment:
Wait, so this still *works* for `old_py3`?

Reply:
my tests show it "working" (completes w/o error) on focal with 3.8 - which was the last python that could install pip from the normal bootstrap url if that's interesting.  Although when I tested bionic I think this part "worked" there too; subject to the same site-packages->dist-packages python setup.py develop fix.

I'd be happy to accept refuting evidence that it does NOT work with bento/ubuntu-20.04 - but my testing suggested it did not exit non-zero and functests passed if that's what you're asking?

Reply:
It just feels really weird that we'd **both** use `pip install -e ...` **and** `python setup.py develop ...` for some boxes; I guess I should try it out myself without one or the other and see how things break.

Reply:
> It just feels really weird that we'd both use pip install -e ... and python setup.py develop ... for some boxes

YES!  SO WEIRD!

As best I can tell it's a ubuntu py3.8 thing - they have a hacked-up/patched python/distutils that IMHO is just returning the "wrong" answer for `site.sitepackages()` - but... I guess they think they have their reasons to not like you `sudo pip install -e .` into `dist-packages` - using `setup.py develop --install-dir` fixes the `import swift` error that is on ALL focal images since some version of pip started warning that `setup.py develop` w/o `pyproject.toml` is no longer going to be supported :shrug: